### PR TITLE
[FIX] openupgradelib: pass encoding in _get_existing_records

### PR DESCRIPTION
**Pull request by mreficent**

### CODE REVIEW

**Comment by mreficent** [comment]
@pedrobaeza And this one?

Remember the error is this line:

```yml
File "/home/travis/virtualenv/python3.5.6/lib/python3.5/site-packages/openupgradelib/openupgrade.py", line 321, in yield_element
    yield StringIO(etree.tostring(path))
TypeError: initial_value must be str or None, not bytes
```

So if this is not the correct fix, it should be similar, for example with a different enconding (unicode, str) or passing other attributes like pretty_print :S

**Comment by pedrobaeza** [comment]
Can you use temporarily this branch in OpenUpgrade PR to check if this works?

**Comment by mreficent** [comment]
Done in https://github.com/OCA/openupgradelib/pull/142.

Seems it was not the correct fix. Any suggestion to try?

**Comment by mreficent** [comment]
@hbrunn your fix in https://github.com/OCA/openupgradelib/commit/522665b2762ed156f8edc258f0ae257428ff332f is broken in python3 due to https://github.com/odoo/odoo/commit/2e6a589f417af48e3322955ffff6e94b6d8a24e3#diff-17401a57dd75921810e818697192c7e6R893 because StringIO record doesn't have name attribute.

Could you help to fix your hack, please? 🙏

**Comment by mreficent** [comment]
@pedrobaeza @hbrunn I wasn't able to fix this :(

File: openupgradelib/openupgrade.py [hunk 1]
```diff
@@ -318,7 +318,7 @@ def yield_element(node, path=None):
                 )
                 if not cr.rowcount:
                     return
-            yield StringIO(etree.tostring(path))
+            yield StringIO(etree.tostring(path, encoding='unicode'))
```
Comment (hbrunn) [inline openupgradelib/openupgrade.py]:
```suggestion
            result = StringIO(etree.tostring(path, encoding='unicode'))
            result.name = None
            yield result
```

Comment (mreficent) [reply openupgradelib/openupgrade.py]: Oh, amazing, thanks 🤯

**Comment by mreficent** [comment]
NEW ERROR:

```yml
$ ./openerp-server -d testdb -i openupgradelib_tests --addons-path addons,/home/travis/build/OCA/openupgradelib/tests/addons --stop-after-init
Traceback (most recent call last):
  File "./openerp-server", line 42, in <module>
    import openerp
  File "/home/travis/openupgrade/openerp/__init__.py", line 28, in <module>
    import addons
  File "/home/travis/openupgrade/openerp/addons/__init__.py", line 38, in <module>
    from openerp.modules import get_module_resource, get_module_path
  File "/home/travis/openupgrade/openerp/modules/__init__.py", line 28, in <module>
    import openerp.modules.graph
  File "/home/travis/openupgrade/openerp/modules/graph.py", line 32, in <module>
    import openerp.osv as osv
  File "/home/travis/openupgrade/openerp/osv/__init__.py", line 22, in <module>
    import osv
  File "/home/travis/openupgrade/openerp/osv/osv.py", line 28, in <module>
    import orm
  File "/home/travis/openupgrade/openerp/osv/orm.py", line 57, in <module>
    import fields
  File "/home/travis/openupgrade/openerp/osv/fields.py", line 44, in <module>
    import openerp.tools as tools
  File "/home/travis/openupgrade/openerp/tools/__init__.py", line 25, in <module>
    from misc import *
  File "/home/travis/openupgrade/openerp/tools/misc.py", line 60, in <module>
    import openerp.pooler as pooler
  File "/home/travis/openupgrade/openerp/pooler.py", line 28, in <module>
    from openerp.modules.registry import RegistryManager
  File "/home/travis/openupgrade/openerp/modules/registry.py", line 28, in <module>
    import openerp.sql_db
  File "/home/travis/openupgrade/openerp/sql_db.py", line 43, in <module>
    from psycopg2.psycopg1 import cursor as psycopg1cursor
ImportError: No module named psycopg1
```

**Comment by mreficent** [comment]
@StefanRijnhart could you check this last error? It's from the openupgradelib tests

**Comment by StefanRijnhart** [comment]
Psycopg1 was dropped in psycopg2 2.8.0 so there is a missing pin on 2.7.7 somewhere.

**Comment by mreficent** [comment]
Ok, fixed. thanks

@pedrobaeza ?

**Comment by pedrobaeza** [comment]
Merging this as we have Travis in red blocking other PR

**Pull request merged**